### PR TITLE
[NCL-7050] Improve ActionButton to make it more generic

### PR DESCRIPTION
**Pull request by jakubvanko**

### CODE REVIEW

File: src/components/ActionButton/ActionButton.tsx [hunk 1]
```diff
@@ -1,42 +1,51 @@
...
+  quality: <TagIcon />,
+  external: <ExternalLinkAltIcon />,
+  close: <LockIcon />,
+  mark: <FlagIcon />,
+} as const;
 
 export interface IActionButtonProps {
-  actionType: 'create' | 'edit' | 'delete';
+  iconType: keyof typeof iconDictionary;
+  children?: React.ReactNode;
```
Comment (matedo1) [inline src/components/ActionButton/ActionButton.tsx:27]: use `React.PropsWithChildren` instead, see for example https://github.com/project-ncl/pnc-web-ui-react/blob/main/src/components/EmptyStates/StateCard.tsx#L18

File: src/components/ProjectDetailPage/ProjectDetailPage.tsx [hunk 2]
```diff
@@ -43,15 +43,27 @@ export const ProjectDetailPage = () => {
...
           <Card>
             <CardBody>
               <AttributesItems attributes={attributes} />
             </CardBody>
           </Card>
         </div>
-        <SectionHeader actions={<ActionButton action={() => console.log('Not implemented yet!')} actionType="create" />}>
+        <SectionHeader
+          actions={
+            <ActionButton action={() => console.log('Not implemented yet!')} iconType="create">
```
Comment (matedo1) [inline src/components/ProjectDetailPage/ProjectDetailPage.tsx:62]: Actually Project Create page is already implemented: `/projects/create`

**Review by matedo1 — CHANGES_REQUESTED** [review]

File: src/components/ProjectDetailPage/ProjectDetailPage.tsx (hunk 2 above)
Comment (jakubvanko) [reply src/components/ProjectDetailPage/ProjectDetailPage.tsx:62]: I think we need to use a link to a BuildConfig create page tho.

Comment (matedo1) [reply src/components/ProjectDetailPage/ProjectDetailPage.tsx:62]: you are right! Deleting this comment, as it's not valid

**Review by matedo1 — COMMENTED** [review]
Could you try to add also support for a button without the icon?

**Pull request merged**